src/impy/iamdata/iamdata_content.json

```diff
@@ -0,0 +1,237 @@
+{
+    "Pythia8": {
+        "data_files": [
+            "Pythia8/xmldoc",
+            "Pythia8/pdfdata",
+            "Pythia8/xmldoc/ProgramFlow.xml",
+            "Pythia8/xmldoc/Welcome.xml",
+            "Pythia8/xmldoc/RIVETusage.xml",
+            "Pythia8/xmldoc/RopeHadronization.xml",
+            "Pythia8/xmldoc/BeamParameters.xml",
+            "Pythia8/xmldoc/ProcessSelection.xml",
+            "Pythia8/xmldoc/Parallelism.xml",
+            "Pythia8/xmldoc/EventInformation.xml",
+            "Pythia8/xmldoc/Tunes.xml",
+            "Pythia8/xmldoc/TimelikeShowers.xml",
+            "Pythia8/xmldoc/VinciaEW.xml",
+            "Pythia8/xmldoc/MatchingAndMerging.xml",
```

Comment:
Is there an advantage in having this as a json file? It is simpler to just make this info a dict in a Python file.

Reply:
Here is another thing that I don't like about this. We are again putting all information for all models in one file. This is not a good data organisation. When I deal with Pythia8, I should find all the information related to making Pythia8 work in the pythia8.py module. This includes the list of data paths that Pythia8 expects. It is not necessary to list all the files inside xmldoc and pdfdata. Important is that xmldoc and pdfdata are extracted from the zip file. We can expect that all files inside those folders are there.

Reply:
So, please do not put this data into a json file. Split this data up and put it as literals into the respective model files as Python dicts. Or better, define a dataclass for this in util.py and use that instead of a dict.

Reply:
We don't need the checksum either, as I have explained below in my comment. When we do operations that need to be "atomic", we perform the operation that takes a while on a fake path name first and then rename the fake path to the final path in the last step, which is super fast. That's how it is done, no checksums required.

Reply:
> Is there an advantage in having this as a json file?

The idea was to track all data files belonging to particular model. If, for some reasons, a file is accidentally deleted or corrupted, it will be automatically restored by downloading and unpacking zip file.

Another point, data files could be changed or additional files could be added  in the future. Then another version of zip files should be prepared and uploaded. To check that the list of files in json and zip files are coincide (the same version) I put checksum of zip files in json. If one needs to make sure that the zip file content is the same as in the list, the check of checksum could be turned on (by default it is off). The checksum is not used for verification whether zip file downloaded or not. Although, it is done only once only after downloading. As downloading is a long operation another ms or so for checksum will not be noticeble. In usual mode, only the existence of the files from the list is checked. No checksums.

json and zip files are created simulatenously by `_create_iamdata_content`. I automatized this task in case it will be needed in the future, and because there are a lot of data files. It is problematic to add files to any "model".py file automatically.

> we perform the operation that takes a while on a fake path name first and then rename the fake path

It is cool trick, and it would be enough if we need just download a file and then check the existence of the same file in the future. Initially I implemented it. But soon I realized there is not need in it. We download zip file, but check the existence of unzipped files. Moreover, if the file is corrupted it will not be recognized as zip file (in this case exception will be raised) and it will not be possible to unpack it.

Reply:
> The idea was to track all data files belonging to particular model. If, for some reasons, a file is accidentally deleted or corrupted, it will be automatically restored by downloading and unpacking zip file.

Files are not accidentally corrupted and no one deletes files in their Python library path. This is preparing for an eventuality that does not happen in practice. If a feature is not needed, it should not be implemented to reduce maintaince cost downstream. If you adhere to this principle now, you will thank me later.

> Another point, data files could be changed or additional files could be added in the future.

That's also not happening and if it does, we just change the path in `iamdata`. For example, for now Pythia-8.3 uses a folder called `Pythia8`. If it turns out that Pythia-8.4 requires different files, we introduce a folder Pythia84. When you update impy locally, the system will then automatically download the required files next time Pythia8 is used. Again, we don't need checksums to address this. We can add some cleanup code that deletes older Paths in `iamdata` that become obsolete after an impy upgrade if that's an issue.

> json and zip files are created simulatenously by _create_iamdata_content. I automatized this task in case it will be needed in the future, and because there are a lot of data files. It is problematic to add files to any "model".py file automatically.

We don't need to add files automatically, because the list of files is very static. It is not a problem at all to put that static information into the "model".py file. Encapsulation is an important design principle and you are breaking it here.

> It is cool trick, and it would be enough if we need just download a file and then check the existence of the same file in the future. Initially I implemented it. But soon I realized there is not need in it. We download zip file, but check the existence of unzipped files. Moreover, if the file is corrupted it will not be recognized as zip file (in this case exception will be raised) and it will not be possible to unpack it.

The only possible time where the download could be corrupted if because an impatient user hits ctrl+C during download. This trick which is widely used by essentially any downloader (firefox...) can recover from this issue. The logic to do this can be encapsulated in CachedPath, so it is completely transparent for us.